src/components/layout/AddToWishlistButton/AddToWishlistButton.js

```diff
@@ -4,49 +4,39 @@ import axiosInstance from '../../../apis/axiosClient'; // axios instance
 import { AuthStatus, useAuth } from '@akosasante/react-auth-context';
 import { useNavigate } from 'react-router-dom';
 
-const wishlistEndpoint = '/wishlist';
-
-const AddToWishlistButton = ({ album }) => {
+const AddToWishlistButton = ({ album, wishListId }) => {
   const [isAdded, setIsAdded] = useState(false);
   const { status } = useAuth();
   const navigate = useNavigate();
   const isUserLoggedIn = status === AuthStatus.LoggedIn;
 
-  const fetchOrCreateWishlist = async () => {
-    try {
-      const response = await axiosInstance.post(wishlistEndpoint);
-      return response.data.wishlist._id; // Return the wishlist ID
-    } catch (error) {
-      console.error('Error creating wishlist:', error);
-    }
-  };
-
   useEffect(() => {
     const storedAddedStatus =
-      JSON.parse(localStorage.getItem(`addedStatus-${album._id}`)) || false;
+      JSON.parse(localStorage.getItem(`wishListAlbums-${album._id}`)) || false;
     setIsAdded(storedAddedStatus);
   }, [album._id]);
 
   const addToWishlist = async () => {
-    const wishlistId = await fetchOrCreateWishlist(); // Fetch the wishlist ID
     try {
       await axiosInstance.patch(
-        `${wishlistEndpoint}/${wishlistId}/add_album/${album._id}`
+        `${wishlistEndpoint}/${wishListId}/add_album/${album._id}`
       );
-      localStorage.setItem(`addedStatus-${album._id}`, JSON.stringify(true));
+      localStorage.setItem(`wishListAlbums-${album._id}`, JSON.stringify(true));
```

Comment:
If you end up making the localstorage field an object, this will need to be updated to take that into account:

```suggestion
      const wishlistAlbums = JSON.parse(localStorage.getItem('wishlistAlbums'))
      wishlistAlbums[album._id] = album;
      localStorage.setItem("wishlistAlbums", JSON.stringify(wishlistAlbums));
```